File: Health-Declaration-Frontend/src/app/model/response/PageResponse.ts
export class PageResponse<T> {
    page: number;
    size: number;
    totalElements: number;
    totalPages: number;
    data: T[];

    constructor(page: number, size: number, totalElements: number, totalPages: number, data: T[]) {
        this.page = page;
        this.size = size;
        this.totalElements = totalElements;
        this.totalPages = totalPages;
        this.data = data;
    }
}

File: Health-Declaration-Backend/src/main/java/com/healthdeclaration/model/response/PageResponse.java
package com.healthdeclaration.model.response;

import lombok.Builder;
import lombok.Getter;

import java.io.Serializable;

@Getter
@Builder
public class PageResponse<T> implements Serializable {
    private int page;
    private int size;
    private long totalElements;
    private long totalPages;
    private T data;
}
